buffer/src/main/java/io/netty/buffer/FixedCompositeByteBuf.java

```diff
@@ -39,10 +39,10 @@
     private final int capacity;
     private final ByteBufAllocator allocator;
     private final ByteOrder order;
-    private final Object[] buffers;
+    private final ByteBuf[] buffers;
     private final boolean direct;
 
-    FixedCompositeByteBuf(ByteBufAllocator allocator, ByteBuf... buffers) {
+    FixedCompositeByteBuf(ByteBufAllocator allocator, boolean copy, ByteBuf... buffers) {
```

Comment:
As an alternative, why not make this never copy, and have callsites copy when they need to?  They have to know anyways in order to set `copy` properly.